src/sentry/workflow_engine/endpoints/serializers.py

```diff
@@ -145,3 +147,44 @@ def serialize(self, obj: Detector, attrs: Mapping[str, Any], user, **kwargs) ->
             "dataSources": attrs.get("data_sources"),
             "conditionGroup": attrs.get("condition_group"),
         }
+
+
+class ErrorDetectorSerializer(DetectorSerializer):
+    def get_attrs(
+        self, item_list: Sequence[Detector], user, **kwargs
+    ) -> MutableMapping[Detector, dict[str, Any]]:
+        attrs = super().get_attrs(item_list, user)
+
+        # collect project options to serialize
+        filtered_item_list = [item for item in item_list if isinstance(item, ErrorDetector)]
+        project_ids = [item.project_id for item in filtered_item_list]
+        project_option_keys = filtered_item_list[0].project_options_config.keys()
+
+        project_options_list = list(
+            ProjectOption.objects.filter(key__in=project_option_keys, project__in=project_ids)
+        )
+
+        configs: dict[int, dict[str, Any]] = defaultdict(dict)
+        for option in project_options_list:
+            configs[option.project_id][option.key] = option.value
+
+        for item in filtered_item_list:
+            if not item.project_id:
+                continue
+            for key in project_option_keys:
+                attrs[item][key] = configs[item.project_id][key]
+
+        return attrs
+
+    def serialize(self, obj: Detector, attrs: Mapping[str, Any], user, **kwargs) -> dict[str, Any]:
+        return {
```

Comment:
fyi - it might be nice to call the parent serialize here so we can reuse the data shape in the API. ideally we have a standardized detector serialized model as well for consistency in the UI.